src/os/bluestore/BlueStore.h

```diff
@@ -1649,6 +1671,36 @@ class BlueStore : public ObjectStore,
       modified_objects.insert(o);
     }
 
+    void note_new_object(OnodeRef &o) {
+      auto [_, ok] = onode_to_offset_map.emplace(
+	  std::pair<OnodeRef, std::vector<int64_t>>(o, {o->get_ondisk_starting_offset()}));
+      ceph_assert(ok);
+    }
+
+    void note_updated_object(OnodeRef &o, int64_t prev_offset) {
+      int64_t new_offset = o->get_ondisk_starting_offset();
+      auto it = onode_to_offset_map.find(o);
+      if (it == onode_to_offset_map.end()) {
+	auto [_, ok] = onode_to_offset_map.emplace(
```

Comment:
it's possible to use emplace to check if onode is present in the map. And hence avoid find() call above. 
From map::emplace spec:
"
Return value
Returns a pair consisting of an iterator to the inserted element, or the already-existing element if no insertion happened, and a bool denoting whether the insertion took place (true if insertion happened, false if it did not).
"

Reply:
Done.